src/login.rs

```diff
@@ -248,9 +249,58 @@ impl<S: TokenStorage> LoginCredentials for RefreshingLoginCredentials<S> {
                 .map_err(RefreshingLoginError::UpdateError)?;
         }
 
+        let mut current_login = self.user_login.lock().await;
+
+        let login = match &*current_login {
+            Some(login) => login.clone(),
+            None => {
+                let response = self
+                    .http_client
+                    .get("https://api.twitch.tv/helix/users")
+                    .header("Client-Id", &self.client_id)
+                    .bearer_auth(&current_token.access_token)
+                    .send()
+                    .await
+                    .map_err(RefreshingLoginError::RefreshError)?;
+
+                log::info!("Fetching username: {}", response.status());
+
+                let users_response = response
+                    .json::<UsersResponse>()
+                    .await
+                    .map_err(RefreshingLoginError::RefreshError)?;
+
+                // If no users are specified in the query, the API reponds with the user of the bearer token.
+                let user = users_response.data.into_iter().next().unwrap();
+
+                log::info!("Current username: {}", &user.login);
```

Comment:
Your wording of "Current username" gave me an idea: 
```suggestion
                # TODO Have the fetched login name expire automatically to be resilient to bot's namechanges
                # should then also automatically reconnect all connections with the new username, so the change
                # will be a little more complex than just adding an expiry to this logic here.
                log::info!("Fetched login name `{}` for provided auth token", &user.login);
```

Reply:
This is an interesting idea, but personally I think that bot name changes are a rare enough occurrence that it's reasonable to expect that you have to restart the bot when you change its name.